### PR TITLE
Support assignments with no submission types

### PR DESCRIPTION
**Pull request by woodie**
The `canvas_cc` gem does not enforce that `submission_types` is set for assignments. These courses with no `submission_types` can be imported into Canvas and work properly, but fail to render in the CCV.

My previous work-around with `canvas_cc` was to always set `assignment.submission_types` to something.
```rb
assignment = CanvasCc::CanvasCC::Models::Assignment.new
assignment.submission_types = ['n/a'] # this must be included currently
```
### In Canvas
We always see the `Submitting` label, which is unpopulated for unknown types.

in_canvas

### In the CCV
We can now have similar behavior by 1) rendering the `Submitting` label unless and submissionFormats is null and 2) returning the NONE type when submissionTypes is null.

in_ccv

### Testing
I've created an `imscc` file with the `canvas_cc` gem [here](https://github.com/woodie/sample-course/blob/main/canvas_cc_gem_course.rb) which has been added for testing.

### CODE REVIEW

**Comment by woodie** [comment]
I need to do additional testing and make a change.

**Comment by woodie** [comment]
I've generated a new course with the `canvas_cc` gem [here](https://github.com/woodie/sample-course/blob/main/canvas_cc_gem_course.rb). Previously, I was creating course content by hand, but was way too difficult to do it that way with this issue. 

I can unzip this new course in the repo if you prefer that. I can probably put the LaTeX samples into this new course to have fewer binary `imscc` files. 

Thanks @akozma89

File: src/utils.js [hunk 1]
```diff
@@ -397,6 +397,8 @@ function isNot(type) {
 }
 
 export function generateFriendlyStringFromSubmissionFormats(submissionType) {
+  if (submissionType === null) return null;
```
Comment (akozma89) [inline src/utils.js:400]: Shouldn't this return `submissionTypeLabels[submissionTypes.NONE]`?

Comment (woodie) [reply src/utils.js:400]: That was my initial thought, but using NONE it renders this:

**Submitting**: Nothing

IMHO we don't want to declare that this assignment has nothing to submit, maybe we just don't know that is submitted, and we what to not render the `Submitting` section. This is what we get when passing an unknown type, like 'n/a'.

File: src/AssignmentBody.js [hunk 2]
```diff
@@ -32,20 +32,19 @@ export default class AssignmentBody extends PureComponent {
...
-                  <Trans>Submitting</Trans>
-                </span>
-                :{" "}
-                <span className="submission-format">
-                  {this.props.submissionFormats}
-                </span>
-              </div>
-            </React.Fragment>
-          )}
+        {this.props.submissionFormats !== null && (
```
Comment (akozma89) [inline src/AssignmentBody.js:35]: I'm trying to understand this change. Do you think it's necessary to do?
If necessary, a simple `this.props.submissionFormats &&` won't do the trick?

Comment (woodie) [reply src/AssignmentBody.js:35]: This change is optional (as the assignment body is displayed without it) but it provides the same behavior as Canvas. With an unknown submissionType, `submissionFormats` is an empty string. With this change, the "Submitting" label is rendered with nothing after, as in Canvas. Without this change, the empty string is evaluated as false, and the "Submitting" label does not render in the CCV.

Comment (woodie) [reply src/AssignmentBody.js:35]: For consistency, line 35 could use `!=` like  line 49 instead of `!==`.

**Review by akozma89 — APPROVED** [review]
Looks good to me.

**Review by subpardaemon — APPROVED** [review]

**Review by subpardaemon — APPROVED** [review]

**Review by subpardaemon — APPROVED** [review]

**Review by subpardaemon — APPROVED** [review]

**Pull request merged**